cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -483,8 +484,26 @@ def create(self, catalog_item_class, *args, **kwargs):
         Returns:
             An instance of catalog_item_class
         """
+        provisioning_entry_point = kwargs.get("provisioning_entry_point", None)
+        reconfigure_entry_point = kwargs.get("reconfigure_entry_point", None)
+        retirement_entry_point = kwargs.get("retirement_entry_point", None)
+        kwargs.pop("provisioning_entry_point")
+        kwargs.pop("reconfigure_entry_point")
+        kwargs.pop("retirement_entry_point")
         cat_item = self.instantiate(catalog_item_class, *args, **kwargs)
-        view = navigate_to(cat_item, 'Add')
+        view = navigate_to(cat_item, "Add")
+        if provisioning_entry_point:
+            view.basic_info.field_entry_point.fill(provisioning_entry_point)
+            view.basic_info.modal.cancel.click()
```

Comment:
What is the purpose of this cancel click?

This still bakes view behavior into the `Entity.create` method, instead of defining that behavior in the view's fill methods.

Separating the entry point fields is good - but they still should be attributes of the catalog item entity class, and included in the kwargs used to instantiate a catalog item.

The changes you've implemented are an improvement, but in my view don't go far enough to correctly model the behavior of the MIQ entity and the MIQ views.

This type of behavior should be part of the `widget.after_fill` method of the widgets used for these entry point fields.

```python
## double check me on the kwargs to after_fill here, writing as example not as runnable code

class FieldEntryPoint(Input):
    def after_fill(self, was_changed):
        if was_changed:
            self.parent_view.modal.cancel.click()
        else:
            pass
```

The above widget would be used for `BasicInfoForm` widgets for the entry points.

Then you can just have a single call to `view.fill(cat_item.fill_dict)` that will also fill the entry point fields and click modal cancel buttons where appropriate.

If there is a need to fill entry points before the rest of the form, this can be achieved with `BasicInfoForm.before_fill` implementation.

Reply:
@mshriver:  I am suspect about this  implementation; please clear my doubts. 

 `Input` is  `widget` and `after_fill` is `View` class method. `Widget` class is  used to create `Input` widget. `Widget` class is parent for `View` class as per inheritance. So as per me this will not work.

`fill` method call will point `Input` widget `fill` method; not `View` class `fill`.  so as per me `aftert_fill` method will not to call. This is magic method for  `Views` not for `Widget`.
I will vote for simple `fill` method overwrite  in this case. 

```
class FieldEntryPoint(Input):
    def fill(self, value):
        current_value = self.value
        if value == current_value:
            return False
        self.browser.click(self)
        self.browser.clear(self)
        self.browser.send_keys(value, self)
        self.parent_view.modal.cancel.click()
        return True
```

Reply:
@digitronik you're absolutely right on widget vs view here, good call. Made comment on the new code, I think we can reduce duplication, but this is the correct approach, and the in-line comments in the current commit are good.